experimental/dds/tree2/src/simple-tree/treeListNode.ts

```diff
@@ -0,0 +1,59 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { AllowedTypes } from "../feature-libraries";
+import { InsertableTreeNodeUnion } from "./insertable";
+import { TreeListNodeBase, TreeNodeUnion } from "./types";
+
+/**
+ * A {@link TreeNode} which implements 'readonly T[]' and the list mutation APIs.
+ * @alpha
+ */
+export interface TreeListNode<out TTypes extends AllowedTypes = AllowedTypes>
+	extends TreeListNodeBase<
+		TreeNodeUnion<TTypes>,
+		InsertableTreeNodeUnion<TTypes>,
+		TreeListNode
+	> {}
+
+/**
+ * A {@link TreeNode} which implements 'readonly T[]' and the list mutation APIs.
+ * @alpha
+ */
+export const TreeListNode = {
+	/**
+	 * Wrap an iterable of items to inserted as consecutive items in a list.
+	 * @remarks
+	 * The object returned by this function can be inserted into a {@link TreeListNode}.
+	 * Its contents will be inserted consecutively in the corresponding location in the list.
+	 * @example
+	 * ```ts
+	 * list.insertAtEnd(TreeListNode.inline(iterable))
+	 * ```
+	 */
+	inline: <T>(content: Iterable<T>) => IterableTreeListContent[create](content),
+};
+
+/**
+ * Non-exported symbol used to make IterableTreeListContent constructable only from within this file.
+ */
+const create = Symbol("Create IterableTreeListContent");
+
+/**
+ * Used to insert iterable content into a {@link (TreeListNode:interface)}.
+ * Use {@link (TreeListNodeOld:variable).inline} to create an instance of this type.
```

Comment:
```suggestion
 * Use {@link (TreeListNode:variable).inline} to create an instance of this type.
```
I think?

Reply:
Only if you want it to work in intelisense and error in API extractor.

On a related note this area is going to need some cleanup once we no longer have both the simple-tree and class-tree TreeListNode types colliding, since this is supposed to work with both of them.

Reply:
API extractor strikes again 🙄